amigo_bot.go

```diff
@@ -298,14 +349,19 @@ func doValidate(config Config, db *sql.DB, ws *websocket.Conn, userToken string,
 		m.Text = fmt.Sprintf("Congrats, you found %s!", event)
 	} else {
 		m.Text = fmt.Sprintf("Sorry, that's not right.")
+		if level == 2 {
+			m.Text += fmt.Sprintf(" You have %d tries left.", 10-(count+1))
+		}
 	}
 	m.Channel = channel
 	postMessage(ws, m)
 	log.Printf("doValidate: done (%s)", u.username)
 }
 
 func doTopScores(config Config, db *sql.DB, ws *websocket.Conn, userToken string, channel string) {
-	rows, err := db.Query("select id, name, unix_timestamp(flag1)-unix_timestamp(start) as flag1_time, unix_timestamp(flag2)-unix_timestamp(start) as flag2_time from teams_start_flag1_flag2 where id < 666 and flag1 is not null order by -flag2_time desc, flag1_time")
+	// I'm sorry
+	rows, err := db.Query("select distinct teams.id as id, teams.name as team_name, unix_timestamp(t1.ts)-unix_timestamp(t0.ts) as flag1_time, unix_timestamp(t2.ts)-unix_timestamp(t0.ts) as flag2_time, FLOOR(unix_timestamp(t3.ts)/unix_timestamp(t3.ts)) AS flag3_time, (select count(*) from logs as a where a.team_id=teams.id and level=2 and (event like 'incorrect:%' or event like 'flag 3')) AS flag3_tries from logs left join teams on (teams.id=logs.team_id) left join logs as t0 on (t0.team_id=logs.team_id and t0.event='start') left join logs as t1 on (t1.team_id=logs.team_id and t1.event='flag 1') left join logs as t2 on (t2.team_id=logs.team_id and t2.event='flag 2') left join logs as t3 on (t3.team_id=logs.team_id and t3.event='flag 3') order by (flag3_time * -flag3_tries) desc, -flag2_time desc, flag1_time")
```

Comment:
Does this sort NULLs for flag3 at the bottom? I initially had a bug where it sorted NULLs for flag2 at the top which was weird